velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -69,6 +69,15 @@ Expected<Timestamp> SparkCastHooks::castIntToTimestamp(int64_t seconds) const {
   return castNumberToTimestamp(seconds);
 }
 
+Expected<int64_t> SparkCastHooks::castTimestampToInt(
+    Timestamp timestamp) const {
+  double micros = static_cast<double>(timestamp.toMicros());
```

Comment:
We don't need to cast to double when micros >= 0

Reply:
done